### PR TITLE
Refactor tests in detekt-gradle-plugin

### PR DESCRIPTION
**Pull request by chao2zhang**
1. Moved data classes, DslGradleRunner, and DslTestBuilder under `testkit`
2. Updated because reason in build.gradle.kts
3. Inlined and updated the formatting version under test.

### CODE REVIEW

File: detekt-gradle-plugin/build.gradle.kts [hunk 1]
```diff
@@ -18,7 +18,7 @@ dependencies {
 
     constraints {
         implementation("org.jetbrains.kotlin:kotlin-reflect:1.4.0") {
-            because("com.android.tools.build:gradle:4.0.1 depends on Kotlin 1.3.72 and we should not mix 1.3 and 1.4")
+            because("$androidGradlePlugin depends on Kotlin 1.3.72 and we should not mix 1.3 and 1.4")
```
Comment (BraisGabin) [inline detekt-gradle-plugin/build.gradle.kts:21]: 🤔 how can we be sure that we remove this constraint as soon as we update to `com.android.tools.build:gradle:4.2.0`. Or at least have a hint that we should remove it.

File: detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskDslTest.kt [hunk 2]
```diff
@@ -637,3 +639,5 @@ internal object DetektTaskDslTest : Spek({
         }
     }
 })
+
+const val VERSION_UNDER_TEST: String = "1.15.0"
```
Comment (BraisGabin) [inline detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskDslTest.kt:643]: Can this be private?

**Review by BraisGabin — APPROVED** [review]
Nice!

It will help if, the next time, you move files in one commit and refactor the file in another. This way we can review it commit by commit and see the actual changes in that file. Otherwise git says "this file was removed and this other was added".

Don't change it now! There's no need.

File: detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskDslTest.kt (hunk 2 above)
Comment (chao2zhang) [reply detekt-gradle-plugin/src/test/kotlin/io/gitlab/arturbosch/detekt/DetektTaskDslTest.kt:643]: Good catch

File: detekt-gradle-plugin/build.gradle.kts (hunk 1 above)
Comment (chao2zhang) [reply detekt-gradle-plugin/build.gradle.kts:21]: https://dl.google.com/android/maven2/com/android/tools/build/gradle/4.2.0-beta02/gradle-4.2.0-beta02.pom. I can add a message stating that this line should be removed once we are on AGP 4.2

**Review by BraisGabin — APPROVED** [review]

**Review by schalkms — APPROVED** [review]

**Pull request merged**